src/components/ShareMyLocationButton.js

```diff
@@ -0,0 +1,48 @@
+import { useContext } from 'react';
+import { MapCenterContext } from '../context/MapCenterContext';
+
+const ShareMyLocationButton = ({ disabled = false }) => {
+  const valueCenterMap = useContext(MapCenterContext);
+  const {
+    defaultCenterMap,
+    setUserCenterMap,
+    setNewCenterMap,
+    setTrackingId,
+    userLocationShared,
+    setUserLocationShared,
+    trackingId,
+  } = valueCenterMap;
+
+  const handleSuccess = (position) => {
+    const lat = position.coords.latitude;
+    const lng = position.coords.longitude;
+    const newMapCenter = { lat, lng };
+    setUserCenterMap(newMapCenter);
+
+    if (!userLocationShared) {
+      setNewCenterMap(newMapCenter);
+      setUserLocationShared(true);
+    }
+  };
+  const handleError = () => setUserCenterMap(defaultCenterMap);
+
+  const handleShareLiveLocation = () => {
+    if (navigator?.geolocation) {
+      if (trackingId === null) {
+        const newTrackingId = navigator?.geolocation?.watchPosition(
+          handleSuccess,
+          handleError,
+        );
+        setTrackingId(newTrackingId);
+      }
+    }
+  };
+
+  return (
+    <button onClick={handleShareLiveLocation} disabled={disabled}>
```

Comment:
maybe for the future, we can show an active state and disable state; use the same button to toggle on and off sharing?  I think we can add this to the next user story as it pertains to the ability to stop sharing.

Reply:
Yes, it's a good idea. We can create a user story or a subtask 👍 .